### PR TITLE
Allow clients the ability to set a default local timeout

### PR DESCRIPTION
**Pull request by Kegsay**
Will be used to fix matrix-org/matrix-appservice-irc#328

### CODE REVIEW

File: lib/base-apis.js [hunk 1]
```diff
@@ -63,7 +63,8 @@ function MatrixBaseApis(opts) {
         request: opts.request,
         prefix: httpApi.PREFIX_R0,
         onlyData: true,
-        extraParams: opts.queryParams
+        extraParams: opts.queryParams,
+        localTimeoutMs: opts.localTimeoutMs
```
Comment (richvdh) [inline lib/base-apis.js]: add to the jsdoc pls

File: lib/http-api.js [hunk 2]
```diff
@@ -626,7 +628,7 @@ module.exports.MatrixHttpApi.prototype = {
         var timeoutId;
         var timedOut = false;
         var req;
-        var localTimeoutMs = opts.localTimeoutMs;
+        var localTimeoutMs = opts.localTimeoutMs || this.opts.localTimeoutMs;
```
Comment (richvdh) [inline lib/http-api.js]: fix the jsdoc at line 598 please

File: lib/base-apis.js (hunk 1 above)
Comment (Kegsay) [inline lib/base-apis.js]: Done

File: lib/http-api.js (hunk 2 above)
Comment (Kegsay) [inline lib/http-api.js]: Done

File: lib/http-api.js [hunk 3]
```diff
@@ -596,7 +596,7 @@ module.exports.MatrixHttpApi.prototype = {
      * @param {object=} opts.headers  extra request headers
      *
      * @param {number=} opts.localTimeoutMs client-side timeout for the
-     *    request. No timeout if undefined.
+     *    request. Default timeout if undefined.
```
Comment (richvdh) [inline lib/http-api.js]: s/undefined/falsy/ ?

Comment (Kegsay) [inline lib/http-api.js]: Done

**Pull request merged**